src/plugins/administrative_templates/registry/abstractregistrysource.h

```diff
@@ -48,8 +48,12 @@ class AbstractRegistrySource
     virtual void markValueForDeletion(const std::string &key, const std::string &valueName)           = 0;
     virtual bool undeleteValue(const std::string &key, const std::string &valueName)                  = 0;
     virtual bool isValueMarkedForDeletion(const std::string &key, const std::string &valueName) const = 0;
+    
+    virtual std::vector<std::string> getNonSpecialValueNames(const std::string &key) const = 0;
+    virtual std::vector<std::string> getValueNames(const std::string &key) const           = 0;
+    
+    virtual void markValueNamesInKeyForDeletion(const std::string &key) = 0;
```

Comment:
This function will delete all entries associated with given `key` thus automatically eliminating key from `registry`, and so name of the function should be `markKeyForDeletion`.